### PR TITLE
Mixed-precision result visualizer

### PR DESCRIPTION
**Pull request by ofirgo**
Adding mixed-precision result visualization.
Translating the final bitwidth config (candidate indices) to actual weights and activation quantization bitwidth, printing to logger and adding figures to tensorboard.

### CODE REVIEW

File: model_compression_toolkit/common/graph/base_graph.py [hunk 1]
```diff
@@ -627,3 +627,23 @@ def get_max_candidates_config(self) -> List[int]:
             f"A maximal config candidate must be defined, but some node have multiple potential maximal candidates"
 
         return [lst[0] for lst in max_cfg_candidates]
+
+    def get_final_weights_config(self):
+        """
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/graph/base_graph.py:632]: Please add type hints

File: model_compression_toolkit/common/graph/base_graph.py [hunk 2]
```diff
@@ -627,3 +627,23 @@ def get_max_candidates_config(self) -> List[int]:
...
+        Gets the final number of bits for quantization of each weights' configurable layer.
+
+        Returns: A list of pairs of (node type, nod's weights quantization bitwidth).
+
+        """
+        sorted_conf_weights = self.get_sorted_weights_configurable_nodes()
+        return [(n, n.final_weights_quantization_cfg.weights_n_bits) for n in sorted_conf_weights]
+
+    def get_final_activation_config(self):
+        """
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/graph/base_graph.py:642]: Please add type hints

File: model_compression_toolkit/common/visualization/final_config_visualizer.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    Args:
+        node: A graph node representing a model's layer.
+
+    Returns: The name of the layer to appear in the visualization.
+
+    """
+
+    return node.type.__name__
+
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/visualization/final_config_visualizer.py:34]: Does it work for functional nodes as well? (does tf.add has it?)

File: model_compression_toolkit/common/visualization/final_config_visualizer.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+# ==============================================================================
+from typing import Tuple, List
+from matplotlib import pyplot as plt
+from matplotlib.figure import Figure
+
+from model_compression_toolkit.common import Graph
+
+
+def get_layer_represent_name(node):
+    """
```
Comment (reuvenperetz) [inline model_compression_toolkit/common/visualization/final_config_visualizer.py:23]: Type hints

**Review by reuvenperetz — APPROVED** [review]

File: model_compression_toolkit/common/visualization/final_config_visualizer.py (hunk 3 above)
Comment (ofirgo) [reply model_compression_toolkit/common/visualization/final_config_visualizer.py:34]: I'm not sure, I can check, but anyway we only apply it to nodes that have kernels that we want to quantize (we present the name only for the weights configurable nodes)

Comment (reuvenperetz) [reply model_compression_toolkit/common/visualization/final_config_visualizer.py:34]: If it's relevant only for weights configurable, so let's rename the function name to indicate it. Also, we need to check for example the case of tf.nn.conv2d (which is functional)

**Pull request merged**